src/containers/rbtree/rbtree.h

```diff
@@ -0,0 +1,794 @@
+  }
+
+  if (old_node == old_node->parent_->left_) {
+    old_node->parent_->left_ = new_node;
+  } else {
+    old_node->parent_->right_ = new_node;
+  }
+}
+
+template <typename K, typename T>
+inline void rbtree<K, T>::delete_case1(node* n) {
+  if (n->parent_) {
+    delete_case2(n);
+  }
+}
+
+template <typename K, typename T>
+inline void rbtree<K, T>::delete_case2(node* find) {
+  node* s = sibling(find);
+  if (s && s->mColor == 'R') {
+    find->parent_->mColor = 'R';
+    s->mColor = 'B';
+    if (find == find->parent_->left_) {
+      rotate_left(find->parent_);
+    } else {
+      rotate_right(find->parent_);
+    }
+  }
+  delete_case3(find);
+}
+
+template <typename K, typename T>
+inline void rbtree<K, T>::delete_case3(node* find) {
+  node* s = sibling(find);
+  if (s && find->parent_->mColor == 'B' && s->mColor == 'B' &&
+      (!s->left_ || s->left_->mColor == 'B') &&
+      (!s->right_ || s->right_->mColor == 'B')) {
+    s->mColor = 'R';
+    delete_case1(find->parent_);
+  } else {
+    delete_case4(find);
+  }
+}
+
+template <typename K, typename T>
+inline void rbtree<K, T>::delete_case4(node* n) {
+  node* s = sibling(n);
+  if (s && n->parent_->mColor == 'R' && (s->mColor == 'B') &&
+      (!s->left_ || s->left_->mColor == 'B') &&
+      (!s->right_ || s->right_->mColor == 'B')) {
+    s->mColor = 'R';
+    n->parent_->mColor = 'B';
+  } else {
+    delete_case5(n);
+  }
+}
+
+template <typename K, typename T>
+inline void rbtree<K, T>::delete_case5(node* n) {
+  node* s = sibling(n);
+
+  if (s && s->mColor == 'B') {
+    if ((n == n->parent_->left_) && (!s->right_ || s->right_->mColor == 'B') &&
+        (!s->left_ || s->left_->mColor == 'R')) {
+      s->mColor = 'R';
+      s->left_->mColor = 'B';
+      rotate_right(s);
+    } else if ((n == n->parent_->right_) && (s->left_->mColor == 'B') &&
+               (s->right_->mColor == 'R')) {
+      s->mColor = 'R';
+      s->right_->mColor = 'B';
+      rotate_left(s);
+    }
+  }
+  delete_case6(n);
+}
+
+template <typename K, typename T>
+inline void rbtree<K, T>::delete_case6(node* n) {
+  node* s = sibling(n);
+  s->mColor = n->parent_->mColor;
+  n->parent_->mColor = 'B';
+  if (n == n->parent_->left_) {
+    s->right_->mColor = 'B';
+    rotate_left(n->parent_);
+  } else {
+    s->left_->mColor = 'B';
+    rotate_right(n->parent_);
+  }
+}
+
+template <typename K, typename T>
+inline bool rbtree<K, T>::is_leaf(node* n) {
+  return (n != nullptr && n->left_ == nullptr && n->right_ == nullptr);
+}
+
+template <typename K, typename T>
+typename rbtree<K, T>::node* rbtree<K, T>::minimum(node* n) {
+  if (n != nullptr) {
+    while (n->left_ != nullptr) {
+      n = n->left_;
+    }
+  }
+  return n;
+}
+
+template <typename K, typename T>
+inline typename rbtree<K, T>::node* rbtree<K, T>::maximum(node* n) {
+  while (n->right_) {
+    n = n->right_;
+  }
+  return n;
+}
+
+template <typename K, typename T>
+inline void rbtree<K, T>::fix_insertion(node* pos) {
```

Comment:
"Красно-чёрное дерево" - это **самобалансирующиеся** дерево, следовательно, метод, который вызывает **балансировку** и должен называться "балансировкой", а не "исправление вставки".

```suggestion
inline void rbtree<K, T>::balance_insetion(node* pos) {
```